### ISSUE TITLE
chunk-manifest not generating

### ISSUE THREAD

**Issue by sntshane** [reactql/kit#146]
i got an error: Error: ENOENT: no such file or directory, open '/home/user/reactql/dist/chunk-manifest.json'
 when i run: npm run build-run
But when i run: npm start, everything is working,
I have rolled back on previous git commit when everything worked perfectly but it still shows the same error, both on localhost and herokuapp.
package.json: "chunk-manifest-webpack-plugin": "git+https://github.com/reactql/chunk-manifest-webpack-plugin.git",

**Comment by dmitry** [comment]
@sntshane I had similar issue. You have to add at least one chunk in the code using something like that:

```
return import(/* webpackChunkName: "./test" */ './test').then(() => {
    return 1;
  }).catch(() => 'An error occurred while loading the component');
```

or change this line:
```
const [manifest, chunkManifest] = ['manifest', 'chunk-manifest']
  .map(name => JSON.parse(readFileSync(path.resolve(PATHS.dist, `${name}.json`), 'utf8')));
```
to
```
const [manifest] = ['manifest']
  .map(name => JSON.parse(readFileSync(path.resolve(PATHS.dist, `${name}.json`), 'utf8')));
const chunkManifest = {};
```

**Comment by dmitry** [comment]
I'm just on-boarded and I'm not sure how it was working before. I just can guess it probably was another version of `webpack` or `chunk-manifest-webpack-plugin`.

**Comment by sntshane** [comment]
That makes sense, thank you again! :+1:

**Comment by ronachong** [comment]
dmitry's fix also worked for me. Here's what I replaced

```
const [manifest, chunkManifest] = ['manifest', 'chunk-manifest']
  .map(name => JSON.parse(readFileSync(path.resolve(PATHS.dist, `${name}.json`), 'utf8')));
```

with in case I want to do some chunking in the future:
```
  const [manifest, chunkManifest] = ['manifest', 'chunk-manifest'].map(
    name => {
      const file = path.resolve(PATHS.dist, `${name}.json`);
      try {
        return JSON.parse(readFileSync(file, 'utf8'),)
      } catch (err) {
        if (err.code === 'ENOENT') {
          console.log(`Error reading ${file}:\n`, err.message);
          console.log('If this is for chunk-manifest.json, it probably means no chunking was used to split app into several bundles.')
          console.log('Defaulting to empty {} for manifest.')
          return {}
        } else {
          throw err;
        }
      }
    },
  );
```

**Comment by dmitry** [comment]
@ronachong nice, this should be committed directly to the `kit`.

**Comment by leebenson** [comment]
Apologies for the generic response, but I've released a new [v3.0 kit](https://github.com/leebenson/reactql/releases/tag/3.0.0), which has been re-written from the ground up.

It bumps the ReactQL stack to Webpack 4, React 16.4, Apollo Client 2.0, and adds Styled Components, full SSR (even in development), and has been re-written in Typescript. It's leaner, faster and has been brought completely up-to-date.

If you're still using ReactQL (and I appreciate you may not be, given the time lag in my response), I'd recommend checking out the new kit to see if your issue has been answered. If not, please feel free to post a new issue at https://github.com/leebenson/reactql
